src/software/new_geom/convex_polygon.h

```diff
@@ -0,0 +1,9 @@
+#include "software/new_geom/convex_shape.h"
+#include "software/new_geom/polygon.h"
+
+/**
+ * A polygon that is convex (curved outwards).
+ */
+class ConvexPolygon : public virtual Polygon, public ConvexShape
```

Comment:
What's the reason for virtual inheritance for `Polygon` here? Also, it seems like there are no guarantees being made that a `ConvexPolygon` is actually `convex`, and the `area` method doesn't seem to be implemented for any generic `ConvexPolygon`.

Reply:
Area method does not need to be implemented in ConvexPolygon, it is not virtual in any of the parent classes.

Reply:
If I wanted to create any generic convex polygon, I should be able to calculate its area. As is, I believe I can create any polygon using `ConvexPolygon` class without any checks on if it's actually a convex polygon, and also not be able to calculate its area. The `area` method also seems to already be written as `virtual` in `ConvexShape`.

Reply:
- I don't think we need virtual inheritance here (see my reply to Mat's comment below for when it's appropriate)
- implementing `area()` shouldn't be too bad for convex polygons (see [here](http://www.mathwords.com/a/area_convex_polygon.htm)). It's definitely declared in `ConvexShape`, so it should definitely be implemented here
- We should probably override the `ConvexPolygon` constructor so that we can't just use the `Polygon` constructor to construct a non-convex polygon?

Reply:
I have virtual inheritance here because Rectangle uses Polygon's constructor and cannot do so unless Polygon is either a virtual or direct base of Rectangle. Is there an alternative both of you were thinking of here?

Reply:
Ah, I see. That's all good then.

Reply:
Could we implement the `area()` function for `ConvexPolygon` @scveloso?

Reply:
Added ```area()``` for ```ConvexPolygon``` in latest update. Also removed ```area``` from ```Rectangle``` and made ```ConvexPolygon``` an abstract class.